src/txn/txn_ckpt.c

```diff
@@ -847,8 +851,26 @@ __txn_checkpoint(WT_SESSION_IMPL *session, const char *cfg[])
         WT_ERR(__wt_txn_checkpoint_log(session, full, WT_TXN_LOG_CKPT_START, NULL));
 
     __checkpoint_timing_stress(session);
+    WT_ERR(__checkpoint_apply_to_dhandles(session, cfg, __checkpoint_tree_helper));
 
-    WT_ERR(__checkpoint_apply(session, cfg, __checkpoint_tree_helper));
+    /* Checkpoint the history store file at snapshot isolation but first refresh our snapshot. */
+    __wt_txn_get_snapshot(session);
+
+    /*
+     * Get an LAS dhandle. If the LAS file is opened for a special operation this will return EBUSY
+     * which we treat as an error.
+     */
+    WT_ERR(__wt_session_get_dhandle(session, WT_LAS_URI, NULL, NULL, 0));
+    las_dhandle = session->dhandle;
+
+    /*
+     * TODO: Is is possible that we don't have a LAS file in certain recovery scenarios. As such we
+     * could get a NULL dhandle.
+     */
+    if (las_dhandle != NULL) {
+        WT_WITH_DHANDLE(session, las_dhandle, ret = __wt_checkpoint(session, cfg));
```

Comment:
What is the reason we do `__wt_checkpoint` instead of `__checkpoint_tree_helper` like every other dhandle.

Reply:
Checkpoint tree helper expects certain things like __checkpoint_lock_dirty_tree to have been run on the given dhandle. However we skip the LAS handle in __checkpoint_get_handles as it can set WT_BTREE_SKIP_CKPT on the history store file as the history store file may not have modifications. However as checkpoint runs it'll add modifications to the history store as such we'd not want to skip it. So because __checkpoint_lock_dirty_tree hasn't been called on the LAS handle we need to call it which is done in __wt_checkpoint.